### PR TITLE
feat(Alert)!: Remove deprecated `onClick` prop

### PR DESCRIPTION
**Pull request by mayank99**
Removed `onClick` prop and the test for it. Added `!` to the title to trigger a major release.

Closes part of #843

### CODE REVIEW

**Comment by bentleyvk** [comment]
Do you want to merge this straight to master? Because it will prevent from making any kind of changes and fixes until v2 is released.

**Comment by mayank99** [comment]
> Do you want to merge this straight to master? Because it will prevent from making any kind of changes and fixes until v2 is released.

Changed base to `v2` branch.

File: packages/iTwinUI-react/src/core/Alert/Alert.tsx [hunk 1]
```diff
@@ -59,7 +54,6 @@ export const Alert = (props: AlertProps) => {
     type = 'informational',
     clickableText,
     clickableTextProps,
-    onClick,
```
[the remark this answers is not in the document]
Comment (mayank99) [reply packages/iTwinUI-react/src/core/Alert/Alert.tsx:62]: Updated

**Review by elephantcatdog — APPROVED** [review]
Looks good to me!

**Pull request merged**